### ISSUE TITLE
Connection

### ISSUE THREAD

**Issue by mimihmaponya** [ashokrajar/SOLR-Nagios-Check-Plugin#8]
I keep getting this error " check_solr: error: no such option: -H " when I install to my nagios instance for my solr 8 connection, what could be the issue.

**Comment by ashokrajar** [comment]
@mimihmaponya yes check_solr command doesn't have a option -H. I believe it's a miss configuration in your Nagios configuration.

Please share how you have configured the command in Nagios.

**Comment by mimihmaponya** [comment]
I am currently looking at it and will send it shortly.

**Comment by mimihmaponya** [comment]
This is the error I encounter. 



I added the below command to the command.cfg file

}

define command {
    command_name    check_solr
    command_line    $USER1$/check_solr.py -H $HOSTADDRESS$ -c $ARG1$

}


Also added the following to my host.cfg file

# Define a service to check Solr on the local machine.
# Disable notifications for this service by default, as not all users may have HTTP enabled.

define service {

    use                     local-service           ; Name of service template to use
    host_name               helen
    service_description     Solr
    check_command           check_solr
    notifications_enabled   0
}

**Comment by ashokrajar** [comment]
@mimihmaponya 

Need some tweak to your configuration. Try this,

define command {
    command_name    check_solr
    command_line    $USER1$/check_solr.py -u $ARG1$ -w $ARG2$ -c $ARG3$
}


define service {
    use                     local-service           ; Name of service template to use
    host_name               Helen
    service_description     Solr
    check_command           check_solr!http://localhost:8090/solr!80!90
    notifications_enabled   0
}


You may want to replace http://localhost:8090/solr with your solr server url. And tune the warning and critical percentage value.

**Comment by mimihmaponya** [comment]
@ashokrajar I have tried the above steps and I get the following error now. (No output on stdout) stderr: execvp(/usr/local/nagios/libexec/check_solr.py, ...) failed. errno is 2: No such file or directory)

**Comment by ashokrajar** [comment]
Error is pretty clear. 

Place the check_solr.py in this repo under /usr/local/nagios/libexec/

**Comment by ashokrajar** [comment]
Please try and let me know the outcome. This plugin was written >8 years back and not sure if it works for latest solr version.

Also it was written with Python 2.x which is EOL.

I would be happy to patch the code for latest solr version if it doesn't.

**Comment by mimihmaponya** [comment]
@ashokrajar Correct, I had not placed the file in the specified directory and I rectified that, I would really appreciate your assistance with the file as now I am running the latest python 3 and the current message it produces is this:

 (No output on stdout) stderr: Traceback (most recent call last):
File "/usr/local/nagios/libexec/check_solr.py", line 83, in <module>
if int(solr_doc_stats.stats['numDocs']) >= int(cmd_options.critical_per):
KeyError: 'numDocs'

**Comment by mimihmaponya** [comment]
@ashokrajar Also please bear with me, I am not really familiar with python but I am actually trying to monitor different services such as solr, gitlab and alfresco but I have been struggling for weeks now using Nagios Core.

**Comment by ashokrajar** [comment]
Great. You managed to configure. Not the above error seems like need a code fix.

Can you share what version of Solr are you using ? I can try to give a new release with support for that version in a day or two.

**Comment by mimihmaponya** [comment]
@ashokrajar absolutely. I am running version 8.0.0 of solr.

**Comment by ashokrajar** [comment]
@mimihmaponya Seems like the recent versions or Solr have deprecated the admin stats url and moved to solr metrics API.

So I will need more time to study the new API and make code changes to support it. It many take couple of weeks, as I plan to do it only in my free time.

Mean while you can try to see if any other OpenSource plugin out there which will meet your requirements.
